### PR TITLE
Construct complex_typet in a non-deprecated way [blocks: #3800]

### PR DESCRIPTION
**Pull request by tautschnig**
The default constructor is deprecated.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 823c1c4bf772ee4e5dda8651dd75843075c080e4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98370481

File: src/ansi-c/literals/convert_integer_literal.cpp [hunk 1]
```diff
@@ -173,10 +173,11 @@ exprt convert_integer_literal(const std::string &src)
...
   {
-    complex_typet complex_type;
-    complex_type.subtype()=type;
+    complex_typet complex_type(type);
     result = complex_exprt(
-      from_integer(0, type), from_integer(value, type), complex_type);
+      from_integer(0, type),
+      from_integer(value, type),
+      std::move(complex_type));
   }
```
Comment (kroening) [inline src/ansi-c/literals/convert_integer_literal.cpp:178]: Nest the constructors?

**Review by kroening — APPROVED** [review]

File: src/ansi-c/literals/convert_integer_literal.cpp (hunk 1 above)
Comment (tautschnig) [reply src/ansi-c/literals/convert_integer_literal.cpp:178]: Done.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: fd1d3940f329e5fd2f7cd50fc7f61d799f94a961).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98448294

**Pull request merged**